### PR TITLE
Gp/general refactoring

### PR DESCRIPTION
**Pull request by JackPiri**
Very general refactoring.

### CODE REVIEW

File: src/test/util_tests.cpp [hunk 1]
```diff
@@ -414,21 +414,22 @@ BOOST_AUTO_TEST_CASE(test_FormatSubVersion)
...
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99999, std::vector<std::string>()), std::string("/Test:0.9.99-dev49/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments),  std::string("/Test:0.9.99-beta1(comment1)/"));
     BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99950, comments),  std::string("/Test:0.9.99(comment1)/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99901, comments2), std::string("/Test:0.9.99-beta1(comment1; comment2)/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99924, comments2), std::string("/Test:0.9.99-beta24(comment1; comment2)/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-alpha(comment1; comment2)/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99950, comments2), std::string("/Test:0.9.99(comment1; comment2)/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-beta1(comment1; comment2)/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-beta1(comment1; comment2)/"));
```
Comment (a-petrini) [inline src/test/util_tests.cpp]: Just a minor non-blocking comment: after refactoring, lines 427 and 428 appear to execute the same check

**Review by a-petrini — APPROVED** [review]
Cool PR, thanks! Left a minor non-blocking comment, feel free to ignore!

File: src/test/util_tests.cpp (hunk 1 above)
Comment (JackPiri) [reply src/test/util_tests.cpp]: Yes sure, this is just duplication. Modified accordingly.

Comment (a-petrini) [reply src/test/util_tests.cpp]: Thanks!

**Review by ptagl — APPROVED** [review]

**Pull request merged**